web/src/settings_account.ts

```diff
@@ -279,6 +282,47 @@ export function update_privacy_settings_box(property: PrivacySettingName): void
     settings_components.set_input_element_value($input_elem, user_settings[property]);
 }
 
+export function render_user_timezone_dropdown_widget(): void {
+    const timezone_items = timezones.timezones.map((tz: {name: string; utc_offset: string}) => ({
+        name: `${tz.name} (${tz.utc_offset})`,
+        unique_id: tz.name,
+    }));
+
+    const opts: dropdown_widget.DropdownWidgetOptions = {
+        widget_name: "user_timezone",
+        get_options: () => timezone_items,
+        item_click_callback(event, dropdown, widget) {
+            dropdown.hide();
+            event.preventDefault();
+            event.stopPropagation();
+            const selected = widget.value()!;
+
+            widget.render(selected);
+            settings_ui.do_settings_change(
+                channel.patch,
+                "/json/settings",
+                {timezone: selected},
+                $(".timezone-setting-status").expectOne(),
+                {
+                    error_continuation() {
+                        widget.render(user_settings.timezone);
+                    },
+                },
+            );
+        },
+        $events_container: $("#profile-settings"),
+        default_id: user_settings.timezone,
+        unique_id_type: "string",
+        search_placeholder_text: $t_html({defaultMessage: "Your time zone"}),
+        text_if_current_value_not_in_options: $t_html({
+            defaultMessage: "Select a time zone",
```

Comment:
This should also be using `$t`, not `$t_html`, since the resulting content is managed as text.